lens-cube/src/main/java/org/apache/lens/cube/parse/UnionCandidate.java

```diff
@@ -21,29 +31,145 @@
   String toStr;
   @Getter
   String alias;
+  CubeQueryContext cubeql;
   /**
    * List of child candidates that will be union-ed
    */
   private List<Candidate> childCandidates;
+  @Getter
+  private QueryAST queryAst;
+  private ASTNode innerSelectAST = new ASTNode(new CommonToken(TOK_SELECT));
+
+  private Map<HashableASTNode, ASTNode> innerToOuterASTs = new HashMap<>();
+  private AliasDecider aliasDecider = new DefaultAliasDecider();
```

Comment:
innerSelectAST, innerToOuterASTs and aliasDecider should have method level scope and should not be declared as class members. These fields are only required within the scope of updateAst() method